### ISSUE TITLE
API Gateway X-Ray Configuration

### ISSUE THREAD

**Issue by mhelgeson** [terraform-providers/terraform-provider-aws#5811]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS has recently added support for X-Ray tracing in API Gateway, alas this configuration cannot yet be managed with terraform. I don't really know if the `aws_api_gateway_method_settings` is the best place for this configuartion, but it seemed to make sense to me.

<img width="426" alt="screen shot 2018-09-07 at 1 45 48 pm" src="https://user-images.githubusercontent.com/508598/45234675-74472380-b2a4-11e8-8e35-3af1385ed802.png">

### New or Affected Resource(s)

* aws_api_gateway_method_settings

### Potential Terraform Configuration

```hcl
# possible example configuration...
resource "aws_api_gateway_method_settings" "example" {
  rest_api_id = "${ aws_api_gateway_rest_api.example.id }"
  stage_name  = "example"
  method_path = "*/*"
  settings {
    xray_enabled = true
  }
}
```

### References

* https://aws.amazon.com/blogs/aws/apigateway-xray/

**Comment by szilveszter** [comment]
According to the documentation, this setting applies to an API Gateway stage, so it should be defined on an `aws_api_gateway_stage` resource.

**Comment by bflad** [comment]
There is an open pull request for this functionality here: #5817 (which will likely be split as I am not sure we should support `aws_api_gateway_deployment` directly)

**Comment by bflad** [comment]
The `xray_tracing_enabled` argument has been added to the `aws_api_gateway_stage` resource in master and will release with version 1.36.0 of the AWS provider, likely very shortly. 👍 

The `aws_api_gateway_deployment` discussion/potential enhancement should probably be tracked in a new, separate issue if someone is interested. These additional parameters during a deployment (that also creates a stage) does not fit the Terraform management model well.

**Comment by bflad** [comment]
This has been released in [version 1.36.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1360-september-13-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
